internal/config/profile.go

```diff
@@ -618,13 +618,38 @@ func (p *Profile) load(readEnvironmentVars bool, envPrefix string) error {
 		// ignore if it doesn't exists
 		var e viper.ConfigFileNotFoundError
 		if errors.As(err, &e) {
+			if envPrefix == MongoCLIEnvPrefix {
+				_ = p.copyOldConfig(readEnvironmentVars, envPrefix)
+			}
 			return nil
 		}
 		return err
 	}
 	return nil
 }
 
+func (p *Profile) copyOldConfig(readEnvironmentVars bool, envPrefix string) error {
```

Comment:
[nit] let's not do this here, let's follow a similar approach to what was done for atlas cli, atlas cli will read if a mongocli config exists and copy it to the atlas cli config path, this check is done at run time when calling a command,
Also notice that logic may need to be updated as well to account for two possible paths

Reply:
Do you mean [this](https://github.com/mongodb/mongocli/blob/098509e33af5bba51b4429bcaae7961231c5c15e/cmd/atlas/atlas.go#L59)? 

I initially thought of following the same approach but overall I thought the logic would look a bit complex if I followed that approach.

Since we load the profile at the beginning of the command, we get an error if we update `MongoCLIConfigHome()` ->` Error: credentials not set`. 

Maybe you thought about something else but what I thought was that we would need to create another profile `oldProfile` and a `newProfile` and try loading `newProfile`, then try `oldProfile` if it doesn't exist until one of them exists. If oldProfile exists we return it and then the atlas command will remove the file, making oldProfile invalid and newProfile loadable. So my understanding was that in the end we would need changes in both files and more clutter to make it work. Again this was what I thought as a solution, please let me know if there is a better way I did not see.

Reply:
+1

Reply:
>Do you mean [this](https://github.com/mongodb/mongocli/blob/098509e33af5bba51b4429bcaae7961231c5c15e/cmd/atlas/atlas.go#L59)?

yes

> Since we load the profile at the beginning of the command, we get an error if we update MongoCLIConfigHome() -> Error: credentials not set.

The logic above runs before initialziing the profile, it's actually nicely seen here
```
	cobra.OnInitialize(createConfigFromMongoCLIConfig, initConfig)
```
first copy then load config

> I thought was that we would need to create another profile oldProfile and a newProfile and try loading newProfile, then try oldProfile if it doesn't exist until one of them exists

Too many responsibilities for the profile to have to remember there's actually two files, Andrea and I already went down discussing the approach for the linked piece of code, you can revisit [here](https://github.com/mongodb/mongocli/pull/977#discussion_r804860498)

And just in case the linked discussion/approach keep the existing mongocli config, we don't need that part we can move it without keeping a back up. This is also why profile should not have a fallback logic, the entry point should always make sure the file is in the right place 

@andreaangiolillo you may want to expand your `+1` as it's not clear what idea are you supporting here

Reply:
I added my comment when Bianca's reply wasn’t there 🤷‍♂️ .

I agree that we should use the same logic that we are using with atlascli

Reply:
cool, thanks for the explanation! let me try to draft this